File: react-kanban/src/components/ui/icons.ts
export type FontawesomeIcon =
  | "circlePlus"
  | "download"
  | "edit"
  | "ellipsis"
  | "ellipsisVertical"
  | "plus"
  | "trash"
  | "reset";

export const iconClasses: Record<FontawesomeIcon, string> = {
  circlePlus: "fa-solid fa-circle-plus",
  download: "fa-solid fa-circle-down",
  edit: "fa-solid fa-pen",
  ellipsis: "fa-solid fa-ellipsis",
  ellipsisVertical: "fa-solid fa-ellipsis-vertical",
  plus: "fa-solid fa-plus",
  trash: "fa-solid fa-trash-can",
  reset: "fa-solid fa-arrow-rotate-left",
};

File: leptos-kanban/src/components/ui/icons.rs
#[allow(dead_code)]
#[derive(Clone, Copy)]
pub enum FontawesomeIcon {
    CirclePlus,
    Download,
    Edit,
    Ellipsis,
    EllipsisVertical,
    Plus,
    Trash,
    Reset,
}

impl FontawesomeIcon {
    pub fn as_fontawesome(&self) -> &'static str {
        match self {
            FontawesomeIcon::CirclePlus => "fa-solid fa-circle-plus",
            FontawesomeIcon::Download => "fa-solid fa-circle-down",
            FontawesomeIcon::Edit => "fa-solid fa-pen",
            FontawesomeIcon::Ellipsis => "fa-solid fa-ellipsis",
            FontawesomeIcon::EllipsisVertical => "fa-solid fa-ellipsis-vertical",
            FontawesomeIcon::Plus => "fa-solid fa-plus",
            FontawesomeIcon::Trash => "fa-solid fa-trash-can",
            FontawesomeIcon::Reset => "fa-solid fa-arrow-rotate-left",
        }
    }
}
